ts/environment.ts

```diff
@@ -24,16 +24,42 @@ import { Package, mkShellPackage } from "./package.ts";
 export type Environment = {
   tag: "environment";
   nixExpression: NixExpression;
-  sandboxSetup: NixExpression;
+  setup: Array<{
+    tag: "common" | "sandbox";
+    snippet: NixExpression;
+  }>;
   description?: string;
 
   /**
    * Update the description for this `Environment`
    */
   setDescription: (this: Environment, newDescription: string) => Environment;
-
   /**
-   * Creates a new environment based on this one that includes the specified nix packages.
+   * Appends the given bash snippet to the setup. The setup will be executed
+   * when running anything in an environment:
+   *
+   * - before running a `Check`, - before building a `Package`, - before running
+   * an `Executable` and - when entering an interactive `Environment` with `garn
+   * enter`.
+   */
+  addToSetup(snippet: string | NixExpression): Environment;
+  /**
+   * Similar to `addToSetup`, but only affects the environment for
+   * `Check`s and `Package`s and does *not* affect `Executable`s or the shell
+   * environment you get with `garn enter`.
+   *
+   * (`Check`s and `Package`s are running in sandboxes to make them completely
+   * deterministic. That's why they have to be treated differently here. The
+   * most common example for how the `Environment` for these sandboxes works
+   * differently is, that we have to *copy* the source files into the sandbox,
+   * to allow `Check`s and `Package`s to access them. Whereas with `Executable`s
+   * and `garn enter`, your source files are already available normally through
+   * the file system.)
+   */
+  addToSandboxSetup(snippet: string | NixExpression): Environment;
+  /**
+   * Creates a new environment based on this one that includes the specified nix
+   * packages.
```

Comment:
Here we add `addToSetup` and `addToSandboxSetup` to the `Environment` type. I think the names of those could be better, but I hope the jsdocs explain well, what they do. Any better ideas for naming these?